api/runtime/v1beta2/extensionconfig_types.go

```diff
@@ -179,7 +179,6 @@ type ExtensionHandler struct {
 	// failurePolicy defines how failures in calls to the ExtensionHandler should be handled by a client.
 	// Defaults to Fail if not set.
 	// +optional
-	// +kubebuilder:validation:Enum=Ignore;Fail
 	FailurePolicy *FailurePolicy `json:"failurePolicy,omitempty"`
```

Comment:
Doesn't need to be a pointer because the zero value is not valid. I believe there was an exception for this in the config?

Reply:
The exclude is for DeletePolicy, not FailurePolicy. But I"m going to open a follow-up PR for this independently.

I think the linter does not detect this correctly, but that is also because of the known enum bug. If I add the Enum marker here back I'm getting:
```go
api/runtime/v1beta2/extensionconfig_types.go:183:2: optionalfields: field FailurePolicy is optional and does not allow the zero value. The field does not need to be a pointer. (kubeapilinter)
        FailurePolicy *FailurePolicy `json:"failurePolicy,omitempty"`
        ^
```

Reply:
PR to fix the pointer: https://github.com/kubernetes-sigs/cluster-api/pull/12453